base/src/main/java/org/mozilla/jss/netscape/security/x509/ExtensionsRequested.java

```diff
@@ -160,17 +160,13 @@ private void construct(DerValue dv) throws IOException {
         // now, the stream will be in the correct format
         stream.reset();
 
-        while (true) {
-            DerValue ext_dv = null;
-            try {
-                ext_dv = stream.getDerValue();
-            } catch (IOException ex) {
-                break;
+        DerValue extDv = null;
+        try {
+            while ((extDv = stream.getDerValue()) != null) {
```

Comment:
Can we do `stream.available() > 0` instead, to check if there is anything in the buffer? This would allow you to drop the `try` with the empty `catch` (I'm not a fan of empty catches), as you break out of the loop before you get the `IOException`. I am sure I tried to do that once and @edewata suggested a reason why not to do it, but I can't remember for the life of me what the reason was. Can you please remind me ^_^

Reply:
Ah, or maybe I tried to do the `null` check in the `while` loop... that sounds familiar. Go team concurrent review, like London buses you wait all day for a review and two come at once :business_suit_levitating: :busstop: :bus: :bus:

Reply:
Sorry, I can't remember what we discussed... :)
I don't have any specific objection for using `stream.available()` here, but it might potentially change the behavior. If we're confident enough it won't cause any problems, then sure.

Reply:
My original idea was to keep the exact same behaviour but I agree it was not the best.

Check the null value requires to modify multiple classes but the check with available  should work as expected. The check is done with 1 because a DER Value require at least 2 byte ([documentation](https://letsencrypt.org/docs/a-warm-welcome-to-asn1-and-der/)) and I have verified in the code. So if the DER value is not read it is for some exception that it is propagated.